### DISCUSSION TITLE
Zoom or Zoom Change not getting called

### DISCUSSION THREAD

**Discussion by csensoft1** [stephanrauh/ngx-extended-pdf-viewer#1526] [Q&A] [answered]
The two-way property Zoom is not being set or zoom change event not being called when I click +/- button as well as when I select percentage level on dropdown like 100%, 150%. It works fine if I select page-width and other predefined levels but not percentage levels.
Could you tell me if this is a known issue or if there is some way to fix the issue. 
I am trying to capture this value and use it but I could not. 

Thank you in advance.

**Comment by stephanrauh** [comment]
You've found a bug! 

BTW, your error description is almost, but not entirely correct. On https://pdfviewer.net/extended-pdf-viewer/two-way-binding, two-way binding ist broken if you start with one of the non-numerical options. If you set the zoom level to a percentage in the Angular form (i.e. the [(zoom)] field), two-way binding works until you select one of the non-numerical options again.

Can you confirm this in your application?

**Reply by stephanrauh** [reply]
Version 15.0.0-alpha.8 fixes this bug.

**Comment by csensoft1** [comment]
Hi Stephan, 
Thanks for responding quickly. 
That is indeed correct steps. I have it default to 'page-width' and then I change it through the dropdown to one of the percentage option then it is not propagating the change to component (angular). However I am able to fix this by listening to currentZoomFactor and updating the two-way binding field in the component. After this, when I change zoom through +/- and likely pinch as well (I did not try this), It goes to percentages that are not there in the zoomlevel, like 110, 130 etc.. I am not able to fix the binded property for this. My solution, I felt like hack and also this issue that I need to fix. So thought of checking with you to see if you can suggest any better solution. 

This is my current function for zoom factor. 
```ts
public currentZoomFactorChange(event: any) {
    if(!isNaN(event)) {
    switch(event) {
      case 0.5:
        this.zoomPercentage = '50%';
        break;
      case 0.75:
        this.zoomPercentage = '75%';
        break;
      case 1.0:
        this.zoomPercentage = '100%';
        break;
      case 1.25:
        this.zoomPercentage = '125%';
        break;
      case 1.5:
        this.zoomPercentage = '150%';
        break;     
    }
    } 
  }
```

**Comment by csensoft1** [comment, marked as answer]
This one looks great on the newer version 15.0.0. Thank you Stephan.
